lib/src/react_client/synthetic_event_wrappers.dart

```diff
@@ -4,153 +4,621 @@
+  /// reference to a `SyntheticEvent` within that callback could have been recycled by ReactJS internals.
+  ///
+  /// You can use `persist()` to ensure access to the event properties within the callback as shown in
+  /// the second example below.
+  ///
+  /// __Without persist()__
+  /// ```dart
+  /// void handleClick(SyntheticMouseEvent event) {
+  ///   print(event.type); // => "click"
+  ///
+  ///   setState({}, () {
+  ///     print(event.type); // => null
+  ///     print(event.isPersistent); => false
+  ///   });
+  /// }
+  /// ```
+  ///
+  /// __With persist()__
+  /// ```dart
+  /// void handleClick(SyntheticMouseEvent event) {
+  ///   print(event.type); // => "click"
+  ///   event.persist();
+  ///
+  ///   setState({}, () {
+  ///     print(event.type); // => "click"
+  ///     print(event.isPersistent); => true
+  ///   });
+  /// }
+  /// ```
+  ///
+  /// See: <https://reactjs.org/docs/events.html#event-pooling>
+  @Deprecated('The modern event system does not use pooling. This does nothing.')
   external void persist();
+
+  /// Whether the event instance has been removed from the ReactJS event pool.
+  ///
+  /// > See: [persist]
+  @Deprecated('The modern event system does not use pooling. This always returns true.')
+  external bool isPersistent();
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [ClipboardEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData>
 @JS()
-@anonymous
 class SyntheticClipboardEvent extends SyntheticEvent {
+  /// [SyntheticClipboardEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticClipboardEvent` instead.
+  external factory SyntheticClipboardEvent._();
+
+  /// Holds a [DataTransfer] object, which can be used:
+  ///
+  /// - to specify what data should be put into the clipboard from the cut and copy event handlers, typically with a
+  /// setData(format, data) call;
+  /// - to obtain the data to be pasted from the paste event handler, typically with a getData(format) call.
+  ///
+  /// See: <https://developer.mozilla.org/en-US/docs/Web/API/ClipboardEvent/clipboardData>
   external get clipboardData;
 }
 
+/// A [SyntheticEvent] wrapper that is specifically backed by a [KeyboardEvent].
+///
+/// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent>
 @JS()
-@anonymous
 class SyntheticKeyboardEvent extends SyntheticEvent {
+  /// [SyntheticKeyboardEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticKeyboardEvent` instead.
+  external factory SyntheticKeyboardEvent._();
+
+  /// Returns `true` if the `Alt` (`Option` or `⌥` on OS X) key was active when the key event was generated.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/altKey>
   external bool get altKey;
+
+  /// Returns a [String] representing the character value of the key.
```

Comment:
#nit Same as the above for the usage of "Returns", and also, specifying/referencing the type aren't necessary because that information is already available via the type annotation.

- https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-variable-getter-or-setter-comments-with-noun-phrases
- https://dart.dev/guides/language/effective-dart/documentation#avoid-redundancy-with-the-surrounding-context
- https://dart.dev/guides/language/effective-dart/documentation#prefer-brevity
```suggestion
  /// The character value of the key.
```